server/src/main/java/org/elasticsearch/cluster/coordination/CoordinationState.java

```diff
@@ -0,0 +1,485 @@
+            logger.debug("handleJoin: ignored join because initial configuration not set");
+            throw new CoordinationStateRejectedException("initial configuration not set");
+        }
+
+        boolean added = joinVotes.addVote(join.getSourceNode());
+        boolean prevElectionWon = electionWon;
+        electionWon = isElectionQuorum(joinVotes);
+        logger.debug("handleJoin: added join {} from [{}] for election, electionWon={} lastAcceptedTerm={} lastAcceptedVersion={}", join,
+            join.getSourceNode(), electionWon, lastAcceptedTerm, getLastAcceptedVersion());
+
+        if (electionWon && prevElectionWon == false) {
+            lastPublishedVersion = getLastAcceptedVersion();
+        }
+        return added;
+    }
+
+    /**
+     * May be called in order to prepare publication of the given cluster state
+     *
+     * @param clusterState The cluster state to publish.
+     * @return A PublishRequest to publish the given cluster state
+     * @throws CoordinationStateRejectedException if the arguments were incompatible with the current state of this object.
+     */
+    public PublishRequest handleClientValue(ClusterState clusterState) {
+        if (electionWon == false) {
+            logger.debug("handleClientValue: ignored request as election not won");
+            throw new CoordinationStateRejectedException("election not won");
+        }
+        if (lastPublishedVersion != getLastAcceptedVersion()) {
+            logger.debug("handleClientValue: cannot start publishing next value before accepting previous one");
+            throw new CoordinationStateRejectedException("cannot start publishing next value before accepting previous one");
+        }
+        if (clusterState.term() != getCurrentTerm()) {
+            logger.debug("handleClientValue: ignored request due to term mismatch " +
+                    "(expected: [term {} version >{}], actual: [term {} version {}])",
+                getCurrentTerm(), lastPublishedVersion, clusterState.term(), clusterState.version());
+            throw new CoordinationStateRejectedException("incoming term " + clusterState.term() + " does not match current term " +
+                getCurrentTerm());
+        }
+        if (clusterState.version() <= lastPublishedVersion) {
+            logger.debug("handleClientValue: ignored request due to version mismatch " +
+                    "(expected: [term {} version >{}], actual: [term {} version {}])",
+                getCurrentTerm(), lastPublishedVersion, clusterState.term(), clusterState.version());
+            throw new CoordinationStateRejectedException("incoming cluster state version " + clusterState.version() +
+                " lower or equal to last published version " + lastPublishedVersion);
+        }
+
+        if (clusterState.getLastAcceptedConfiguration().equals(getLastAcceptedConfiguration()) == false
+            && getLastCommittedConfiguration().equals(getLastAcceptedConfiguration()) == false) {
```

Comment:
Tests pass with `if (getLastCommittedConfiguration().equals(getLastAcceptedConfiguration()) == false) {`

Reply:
fixed in 708ae8fd098